### PR TITLE
PB-1242 Add app-origin-version & bug fixes

### PR DESCRIPTION
**Pull request by peteward**
- Fixed a bug I created as part of the plugin review process - I did escaping wrong and it made a mess of redirect and messages when you try to print...
- Fixed the duplicate sends bug

### CODE REVIEW

File: src/Admin/OrderAdminExtension.php [hunk 1]
```diff
@@ -42,16 +42,15 @@ public function addOrderMetaBoxActions($actions)
...
-                $orderAdminEnabled = \WC_Admin_Settings::get_option(Settings::FIELD_ENABLE_ORDER_EXTENSIONS);
+            }
+
+            $orderAdminEnabled = \WC_Admin_Settings::get_option(Settings::FIELD_ENABLE_ORDER_EXTENSIONS);
 
-                if ($orderAdminEnabled && $orderAdminEnabled !== 'no') {
-                    $actions['penny_black_print'] = 'Print via Penny Black';
-                }
+            if ($orderAdminEnabled && $orderAdminEnabled !== 'no') {
+                $actions['penny_black_print'] = 'Print via Penny Black';
```
Comment (peteward) [inline src/Admin/OrderAdminExtension.php:53]: it's possible that you could get an error of "already sent" for an order (which you can't recover from) so I removed the else on this condition to ensure we always have a print option visible

File: src/Admin/OrderAdminExtension.php [hunk 2]
```diff
@@ -117,31 +116,31 @@ public function handleBatchPrintRequest($redirectTo, $action, $postIds)
...
+        $this->message = $message;
+        $this->messageType = 'success';
+        return $this->addNotificationQueryVars($redirectTo);
     }
 
     public function addNotificationQueryVars($location)
     {
-        return esc_url(add_query_arg(['pb_msg_type' => $this->messageType, 'pb_msg' => $this->message], $location));
+        return add_query_arg(
+            ['pb_msg_type' => rawurlencode($this->messageType), 'pb_msg' => rawurlencode($this->message)],
```
Comment (peteward) [inline src/Admin/OrderAdminExtension.php:141]: `esc_url()` was wrong, this creates a whole URL for you, even if what you pass isn't a URL, so it's not usable for URL _parameters_. Switched to just `rawurlencode` and updated all approaches to use this method.

File: src/PennyBlackPlugin.php [hunk 3]
```diff
@@ -43,4 +45,13 @@ public static function createSettingsLink($links)
 
         return $links;
     }
+
+    public static function getVersion(): string
+    {
+        if (!isset(self::$VERSION)) {
+            $pluginData = get_file_data(__DIR__ . '/../woocommerce-pennyblack.php', array('Version' => 'Version'), false);
+            self::$VERSION = $pluginData['Version'];
```
Comment (peteward) [inline src/PennyBlackPlugin.php:53]: this is not wonderful using statics, but the alternative was passing version numbers down through lots of classes or a major restructure which was not appealing.

**Review by jameshalsall — APPROVED** [review]
great

**Pull request merged**